Simplenote/Classes/SPNoteEditorViewController+Extensions.swift

```diff
@@ -463,17 +463,53 @@ extension SPNoteEditorViewController {
 
     @objc
     func ensureEmptyNoteIsDeleted() {
-        guard let note = currentNote else {
-            return
-        }
-
         guard note.isBlank, noteEditorTextView.text.isEmpty else {
             save()
             return
         }
 
         SPObjectManager.shared().trashNote(note)
-        currentNote = nil
+    }
+
+    @objc
+    private func handleTapOnCreateNewNoteButton() {
+        saveIfNeeded()
+
+        if note.isBlank {
+            noteEditorTextView.becomeFirstResponder()
+            return
+        }
+
+        SPTracker.trackEditorNoteCreated()
+
+        let snapshotRect = CGRect(x: 0,
+                                  y: noteEditorTextView.adjustedContentInset.top,
+                                  width: noteEditorTextView.frame.width,
+                                  height: noteEditorTextView.frame.height - noteEditorTextView.adjustedContentInset.top)
+
+        guard let snapshotView = view.resizableSnapshotView(from: snapshotRect, afterScreenUpdates: false, withCapInsets: .zero),
+              let navigationController = navigationController else {
+            return
+        }
+
+        snapshotView.frame.origin.y = noteEditorTextView.adjustedContentInset.top
+
+        navigationController.view.addSubview(snapshotView)
+
+        let viewControllers: [UIViewController] = navigationController.viewControllers.map{
+            if $0 == self {
+                return EditorFactory.shared.build(with: nil)
```

Comment:
I've tested on my X and feels smooth. We **could** maybe consider reusing, but honestly, the code looks wayyyy much cleaner!!

Reply:
We were already creating new editor every time you open a note from the list. This was the last place where we were reusing the editor. I think reusing makes code more complex, so unless there are serious issue with performance, not reusing makes life simpler :-)

Reply:
I've been thinking (for a while!!) on this one, and it's definitely a good call. Simplifies the Editor, quite dramatically.

Thank you!!